### DISCUSSION TITLE
JQ extension to yaml -> JYQ

### DISCUSSION THREAD

**Discussion by ArTaSHseTiN** [stedolan/jq#3077] [Ideas]
Is it possible to extend this program to work with yaml also. We can add a yaml to json converter and vice versa

**Comment by netj** [comment]
fwiw there is python-yq https://kislyuk.github.io/yq/ I've been enjoying.

Don't confuse with https://mikefarah.gitbook.io/yq which is not a simple jq wrapper but an attempt to re-implement jq itself in golang that happens to take yaml as input as well.

I'd be happy to see jq directly handling yaml, but YAML parsing itself brings a lot of complexity. Just using python-yq for YAML I/O as needed might be a healthier setup for jq IMHO.

**Reply by blurayne** [reply]
Python YAML wrapper is slow. Better use fq or oq.

**Comment by wader** [comment]
@blurayne fq 🥳

Also https://github.com/itchyny/gojq and https://github.com/MiSawa/xq can do YAML
